packages/relic/test/headers/special/accept_encoding_header_test.dart

```diff
@@ -0,0 +1,280 @@
+import 'dart:io';
+import 'package:test/test.dart';
+import 'package:relic/src/headers/headers.dart';
+import 'package:relic/src/relic_server.dart';
+import '../headers_test_utils.dart';
+
+void main() {
+  // Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding
+  group('Given an Accept-Encoding header with the strict flag true', () {
+    late RelicServer server;
+
+    setUp(() async {
+      try {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv6, 0);
+      } on SocketException catch (_) {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv4, 0);
+      }
+    });
+
+    tearDown(() => server.close());
+
+    test(
+      'when an empty Accept-Encoding header is passed then it should throw TestInvalidHeaderException',
```

Comment:
Discussion: What we really want to test is that the server rejects the message with a specific error message. Would it be better to expect a 400 with the expected message instead of this exception?

Reply:
Suggestion: We include the message we are expecting in the description.